### PR TITLE
order events in event store

### PR DESCRIPTION
**Pull request by AnthonyBuisset**
- :card_file_box: drop event timestamp default value in event store
- :sparkles: add timestamp in storable event
- :sparkles: use block timestamp to store blockchain events
- :recycle: actually sort events by timestamp before index

### CODE REVIEW

**Review by Bernardstanislas — APPROVED** [review]
Great! Events in order, as it should be :)

File: marketplace-infrastructure/src/database/models/events.rs [hunk 1]
```diff
@@ -1,15 +1,26 @@
...
-#[derive(Insertable, Debug, Serialize, Deserialize)]
+#[derive(Debug, Serialize, Deserialize)]
 #[serde(crate = "rocket::serde")]
 pub struct Event {
 	pub aggregate_name: String,
 	pub aggregate_id: String,
 	pub payload: Value,
 }
 
+#[derive(Insertable, Debug, Serialize, Deserialize)]
```
Comment (ofux) [inline marketplace-infrastructure/src/database/models/events.rs:14]: good to have specific models for creation 👍

File: marketplace-infrastructure/src/database/event_store/event_store.rs [hunk 2]
```diff
@@ -96,7 +97,8 @@ impl<A: NamedAggregate> EventStore<A> for Client {
 			.select(events::payload)
 			.filter(events::aggregate_id.eq(aggregate_id.to_string()))
 			.filter(events::aggregate_name.eq_all(A::name()))
-			.order_by(events::index)
+			.order_by(events::timestamp)
```
Comment (ofux) [inline marketplace-infrastructure/src/database/event_store/event_store.rs:100]: `timestamp` column should be indexed for faster ordering (see https://www.postgresql.org/docs/current/indexes-ordering.html)

**Review by ofux — APPROVED** [review]

File: marketplace-infrastructure/src/database/event_store/event_store.rs (hunk 2 above)
Comment (AnthonyBuisset) [reply marketplace-infrastructure/src/database/event_store/event_store.rs:100]: done

**Pull request merged**